lib/HLSL/DxilLegalizeSampleOffsetPass.cpp

```diff
@@ -112,16 +112,15 @@ bool HasIllegalOffsetInLoop(std::vector<Instruction *> &illegalOffsets,
   return findOffset;
 }
 
-void CollectIllegalOffset(CallInst *CI,
+void CollectIllegalOffset(CallInst *CI, unsigned offsetIdx,
                           std::vector<Instruction *> &illegalOffsets) {
   Value *offset0 =
-      CI->getArgOperand(DXIL::OperandIndex::kTextureSampleOffset0OpIdx);
+      CI->getArgOperand(offsetIdx);
   // No offset.
   if (isa<UndefValue>(offset0))
     return;
 
-  for (unsigned i = DXIL::OperandIndex::kTextureSampleOffset0OpIdx;
-       i <= DXIL::OperandIndex::kTextureSampleOffset2OpIdx; i++) {
+  for (unsigned i = offsetIdx; i <= offsetIdx + 2; i++) {
     Value *offset = CI->getArgOperand(i);
     if (Instruction *I = dyn_cast<Instruction>(offset))
```

Comment:
Shouldn't we be checking for constant offsets out of `[-8..7]` range here as well, and collecting these to a separate list (intrinsic instruction instead of argument, since argument would be constant)?

Reply:
The intent of this pass is to try to make the samples constant if possible. the generation of an error is incidental. I could tack on a check for the ranges here if it would be of benefit, but I'm not sure it would be much different from the validation location.

In fact, the followup change I'm working on has had to revise some of this code more than I originally intended to get errors that are of any use since the existing code only keeps around the offset source, which is often not something with a debug location and not as helpful as the calling function that takes the offsets anyway.